### DISCUSSION TITLE
automating gem releases

### DISCUSSION THREAD

**Discussion by benkoshy** [andrewmcodes/andrewm-codes-website#51] [AMA] [answered]
Hi Andrew

Thank you for your article: https://andrewm.codes/blog/automating-ruby-gem-releases-with-github-actions/

I have a few questions:

-  Do I have to manually update the gem's version in the version.rb file? Or will this be handled automatically by all the actions.
- Also was curious whether you knew anything about how the github actions picks up the version number of the gem and applies a tag onto the git repository (coz the git versions that are tagged on my repository seem to be out of sync) with the version number noted on the `version.rb`)?

Any light you could shed would be most appreciated.

rgds

Ben

**Comment by andrewmcodes** [comment, marked as answer]
> Do I have to manually update the gem's version in the version.rb file? Or will this be handled automatically by all the actions.

It should be automatically updated by the action in the PR it creates with the upcoming launch. In that PR it should be updating the version according to your semantic commits and changelog.

Here is an example from this project: #62. I know it's updating `package.json` instead of `version.rb` in this example but it's the same concept. You tell it what to point to in the action config by specifying `release-type` in the action ([example](https://github.com/andrewmcodes/andrewm-codes-website/pull/56/files#diff-2c84033033d49186c63e6adcd705f63b11ae6814cd76c152c9c486d389fbccf3R13)). For Ruby projects, it will look for `version.rb` based on [this chart](https://github.com/google-github-actions/release-please-action#release-types-supported).

> Also was curious whether you knew anything about how the github actions picks up the version number of the gem and applies a tag onto the git repository (coz the git versions that are tagged on my repository seem to be out of sync) with the version number noted on the version.rb)?

Hmm it should be basing the number off of `version.rb` but it will change it based on your commit messages so anything prefixed with `fix:` would be a patch bump, `feat:` is minor, and if you use a `!` to indicate a breaking change, it will bump a full version number. Hard to say without being able to see what's going on but I did have issues originally (that went away) with packages pre 1.0.0 with Ruby so maybe your issues have resolved in the _reads when this was submitted_ 6 months that I didn't have notifications on for this repo. 

Apologies about the horribly late reply but I'd be happy to look at it with you. I use it on all my projects now and install it on the old one's whenever I return.

**Reply by benkoshy** [reply]
Hey Andrew I appreciate the detailed response. (I had actually forgotten all about it, and regret to not have come back and alerted you to this. i used a regex to extract the version number manually from the ruby / javascript / json files. I ought to circle back given your kind response above). Ben
